File: 13. Roman to Integer/Solution13.java

import java.util.HashMap;

class Solution13 {
    public static int romanToInt(String s) {
        HashMap<Character, Integer> romanMap = new HashMap<Character, Integer>();
        romanMap.put('I', 1);
        romanMap.put('V', 5);
        romanMap.put('X', 10);
        romanMap.put('L', 50);
        romanMap.put('C', 100);
        romanMap.put('D', 500);
        romanMap.put('M', 1000);
        int sum = 0;
        for (int i = 0; i < s.length(); i++) {
            if (i < s.length() - 1 && romanMap.get(s.charAt(i)) < romanMap.get(s.charAt(i + 1))) {
                sum += romanMap.get(s.charAt(i + 1)) - romanMap.get(s.charAt(i));
                i++;
            } else {
                sum += romanMap.get(s.charAt(i));
            }

        }
        return sum;
    }
}

public static void main(String[] args) {
    System.out.println(Solution13.romanToInt("MCMXCIV"));
}

File: 13. Roman to Integer/Solution13.py
class Solution(object):
    def romanToInt(self, s):        
       romanMap = {"I":1, "V":5, "X":10, "L":50,"C":100,"D":500,"M":1000}
       sum = 0
       length = len(s)
       i = 0
       while i < length:
            if i < length - 1 and romanMap[s[i]] < romanMap[s[i+1]]:
                sum += romanMap[s[i+1]] - romanMap[s[i]]
                i += 2
            else:
                sum += romanMap[s[i]]
                i += 1
       return sum
    
c = Solution()
print (c.romanToInt("MCMXCIV"))       